From 780d54720ed6c2686c7cc50f9756f7720e1ff31c Mon Sep 17 00:00:00 2001
From: Jes Sorensen <Jes.Sorensen@redhat.com>
Date: Mon, 1 Sep 2014 09:26:12 -0400
Subject: [md] make sure GET_ARRAY_INFO ioctl reports correct "clean" status

Message-id: <1409563584-9000-71-git-send-email-Jes.Sorensen@redhat.com>
Patchwork-id: 90007
O-Subject: [RHEL7.1 PATCH 70/82] md: make sure GET_ARRAY_INFO ioctl reports correct "clean" status
Bugzilla: 1085530
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

From: NeilBrown <neilb@suse.de>

If an array has a bitmap, the when we set the "has bitmap" flag we
incorrectly clear the "is clean" flag.

"is clean" isn't really important when a bitmap is present, but it is
best to get it right anyway.

Reported-by: George Duffield <forumscollective@gmail.com>
Link: http://lkml.kernel.org/CAG__1a4MRV6gJL38XLAurtoSiD3rLBTmWpcS5HYvPpSfPR88UQ@mail.gmail.com
Fixes: 36fa30636fb84b209210299684e1be66d9e58217 (v2.6.14)
Signed-off-by: NeilBrown <neilb@suse.de>
(cherry picked from commit 9bd359203210efeb5d8f0d81c155079f34b47449)
Signed-off-by: Jes Sorensen <Jes.Sorensen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/md/md.c b/drivers/md/md.c
index abc3bf7..c33e233 100644
--- a/drivers/md/md.c
+++ b/drivers/md/md.c
@@ -5599,7 +5599,7 @@ static int get_array_info(struct mddev * mddev, void __user * arg)
  if (mddev->in_sync)
   info.state = (1<<MD_SB_CLEAN);
  if (mddev->bitmap && mddev->bitmap_info.offset)
-  info.state = (1<<MD_SB_BITMAP_PRESENT);
+  info.state |= (1<<MD_SB_BITMAP_PRESENT);
  info.active_disks  = insync;
  info.working_disks = working;
  info.failed_disks  = failed;
-- 
1.7.1